Packs/AWS-WAF/Integrations/AWS-WAF/AWS_WAF.py

```diff
@@ -0,0 +1,1122 @@
+        that are larger than AWS WAF can inspect
+        text_transformation: The text transformation to perform on the component
+        string_to_match: The string to match to
+        match_type: Which match type should be performed
+
+    Returns:
+        A byte match statement value
+    """
+    web_request_component_object = build_web_component_match_object(web_request_component, oversize_handling)
+    return {
+        'SearchString': string_to_match,
+        'FieldToMatch': {
+            web_request_component: web_request_component_object
+        },
+        'TextTransformations': [
+            {'Priority': 0, 'Type': text_transformation
+             }
+        ],
+        'PositionalConstraint': MATCH_TYPE_TO_POSITIONAL_CONSTRAIN[match_type]}
+
+
+def build_regex_match_statement(web_request_component: str,
+                                oversize_handling: str,
+                                text_transformation: str,
+                                regex_set_arn: str) -> dict:
+    """
+    Creates a byte match statement
+    Args:
+        web_request_component: The web request component to inspect
+        oversize_handling: Which oversize handling to perform on web request contents
+        that are larger than AWS WAF can inspect
+        text_transformation: The text transformation to perform on the component
+        regex_set_arn: The regex set to match to
+
+    Returns:
+        A regex match statement value
+    """
+    web_request_component_object = build_web_component_match_object(web_request_component, oversize_handling)
+    return {
+        'ARN': regex_set_arn,
+        'FieldToMatch': {
+            web_request_component: web_request_component_object
+        },
+        'TextTransformations': [
+            {'Priority': 0, 'Type': text_transformation
+             }
+        ]
+    }
+
+
+def build_string_match_rule_object(args: dict) -> dict:
+    """
+    Creates a string match rule statement object that can be added to a rule
+    Args:
+        args: The command arguments
+
+    Returns:
+        String match rule statement object
+    """
+    string_match_statement: dict = {
+        'Statement': build_string_match_statement(args)
+    }
+    return string_match_statement
+
+
+def build_new_rule_object(args: dict, rule_group_visibility_config: dict,
+                          build_rule_func: Callable[[dict], dict]) -> dict:
+    """
+    Creates a country rule object that can be added to a rule group rules list
+    Args:
+        args: The command arguments
+        rule_group_visibility_config: The rule visibility config
+        build_rule_func: A generic function that builds the statement of the rule
+
+    Returns:
+        Entire rule object
+    """
+    name = args.get('rule_name', '')
+    rule_visibility_config = build_visibility_config_object(
+        metric_name=name,
+        cloud_watch_metrics_enabled=rule_group_visibility_config.get('CloudWatchMetricsEnabled'),  # type: ignore
```

Comment:
why needed `# type: ignore`? And if needed, please specify which error in the ignore

Reply:
```suggestion
        cloud_watch_metrics_enabled=rule_group_visibility_config.get('CloudWatchMetricsEnabled', true),  
```